cpp/dolfinx/fem/Function.h

```diff
@@ -300,8 +300,10 @@ class Function
     xt::xtensor<double, 1> detJ = xt::zeros<double>({1});
 
     // Prepare basis function data structures
-    xt::xtensor<double, 3> basis_reference_values(
-        {1, space_dimension, reference_value_size});
+    xt::xtensor<double, 4> basis_derivatives_reference_values(
+        {1, 1, space_dimension, reference_value_size});
+    auto basis_reference_values = xt::view(basis_derivatives_reference_values,
+                                           0, xt::all(), xt::all(), xt::all());
```

Comment:
A little issue is that this will be copied further down in each loop iteration when passed to `element->transform_reference_basis` since `transform_reference_basis` expects a `xt::xarray`.

Reply:
That is not correct?
Transform reference basis takes an xtensor: https://github.com/FEniCS/dolfinx/blob/13b06ce99e79a6972b55cdc4710df05b4a93d79b/cpp/dolfinx/fem/FiniteElement.cpp#L200-L208
and 
`map_push_forward_m` takes an xtensor: https://github.com/FEniCS/basix/blob/266c9b486d6a8e6d00c118cb05f499690cd98679/cpp/basix/finite-element.h#L317

Reply:
Yes, it takes a `xt::xtensor<double, 3>`, but `xt::view` isn't a `xt::xtensor<double, 3>` hence a copy will be made.